src/components/Trends/TrendDepthToggle/TrendDepthToggle.spec.js

```diff
@@ -0,0 +1,192 @@
+
+const renderComponent = (newFiltersState, newTrendsState) => {
+  merge(newFiltersState, filtersState);
+  merge(newTrendsState, trendsState);
+  const data = {
+    filters: newFiltersState,
+    query: queryState,
+    routes: { queryString: '?fdsafsfoo' },
+    trends: newTrendsState,
+    view: { tab: MODE_TRENDS },
+  };
+  render(<TrendDepthToggle />, { preloadedState: data });
+};
+
+describe('component:TrendDepthToggle', () => {
+  let depthChangedSpy, depthResetSpy;
+  beforeEach(() => {
+    fetchMock.resetMocks();
+
+    depthChangedSpy = jest
+      .spyOn(trendsActions, 'depthChanged')
+      .mockImplementation(() => jest.fn());
+    depthResetSpy = jest
+      .spyOn(trendsActions, 'depthReset')
+      .mockImplementation(() => jest.fn());
+  });
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+  it('does not render on Focus page', () => {
+    fetchMock.mockResponse((req) => {
+      if (req.url.indexOf('API/trends?') > -1) {
+        return Promise.resolve({
+          body: JSON.stringify(trendsResponse),
+        });
+      } else if (req.url.indexOf('API?') > -1) {
+        // this is the list
+        return Promise.resolve({
+          body: JSON.stringify(aggResponse),
+        });
+      }
+    });
+    renderComponent({}, { focus: 'Foo bar' });
+    expect(screen.queryByRole('button', { name: 'Show more' })).toBeNull();
+  });
+
+  it('does not render when lens is Overview', () => {
+    renderComponent({}, { lens: 'Overview' });
+    expect(screen.queryByRole('button', { name: 'Show more' })).toBeNull();
+  });
+
+  it('renders Product view more link', async () => {
+    fetchMock.mockResponse((req) => {
+      if (req.url.indexOf('API/trends?') > -1) {
+        return Promise.resolve({
+          body: JSON.stringify(trendsResponse),
+        });
+      } else if (req.url.indexOf('API?') > -1) {
+        // this is the list
+        return Promise.resolve({
+          body: JSON.stringify(aggResponse),
+        });
+      }
+    });
+
+    renderComponent({}, {});
+
+    await screen.findByRole('button', { name: 'Show more' });
+    expect(
+      screen.getByRole('button', { name: 'Show more' }),
+    ).toBeInTheDocument();
+
+    fireEvent.click(screen.getByRole('button', { name: 'Show more' }));
+    expect(depthChangedSpy).toHaveBeenCalledWith(14);
+  });
+
+  it('renders Product show less link', async () => {
+    fetchMock.mockResponse((req) => {
+      if (req.url.indexOf('API/trends?') > -1) {
+        return Promise.resolve({
+          body: JSON.stringify(trendsResponseDepth14),
+        });
+      } else if (req.url.indexOf('API?') > -1) {
+        // this is the list
+        return Promise.resolve({
+          body: JSON.stringify(aggResponse),
+        });
+      }
+    });
+    renderComponent(
+      {},
+      {
+        lens: 'Product',
+        trendDepth: 14,
+      },
+    );
+
+    await screen.findByRole('button', { name: 'Show less' });
```

Comment:
Can button be used in one variable across both lines 118 and 119?

Reply:
This is intentional, this case, await returns a Promise and we use it here to allow the button to render asynchronously. 

Normally in Cypress tests we're not supposed to be saving the selectors in a variable.  They don't recommend it.

> This is one of the primary reasons why we do not recommend using [variables](https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Variables) within your tests.

https://learn.cypress.io/cypress-fundamentals/understanding-the-asynchronous-nature-of-cypress

Typically I prefer to have the screen.getBy calls repeated explicitly instead of saving it in a variable  because if the DOM somehow changes, each screen.getBy call is a new call to check the document instead of having a saved variable.